cmd/ddev/cmd/config.go

```diff
@@ -33,22 +39,43 @@ var ConfigCommand = &cobra.Command{
 
 		c, err := ddevapp.NewConfig(appRoot, provider)
 		if err != nil {
-			// If there is an error reading the config and the file exists, we're not sure
-			// how to proceed.
-			if c.ConfigExists() {
-				util.Failed("Could not read config: %v", err)
-			}
+			util.Failed("Could not create new config: %v", err)
 		}
 
 		// Set the provider value after load so we can ensure we use the passed in provider value
 		// for this configuration.
 		c.Provider = provider
+		c.Name = siteName
+		c.Docroot = docrootRelPath
 
-		err = c.Config()
-		if err != nil {
-			util.Failed("There was a problem configuring your application: %v\n", err)
-		}
+		if siteName == "" && docrootRelPath == "" && pantheonEnvironment == "" {
+			err = c.PromptForConfig()
+			if err != nil {
+				util.Failed("There was a problem configuring your application: %v\n", err)
+			}
+		} else {
+			// Handle the case where flags have been passed in and we config that way
+			appType, err := ddevapp.DetermineAppType(c.Docroot)
+			if err != nil {
+				fullPath, _ := filepath.Abs(c.Docroot)
+				util.Failed("Failed to determine app Type (drupal7/drupal8/wordpress), your docroot may be incorrect - looking in directory %v (full path: %v), err: %v", c.Docroot, fullPath, err)
```

Comment:
- this could end up being a long message, we might want to consider breaking it to another line.
- i feel like we should avoid displaying "err" to end users. this is already an error message, it seems weird to call out an error inside it. "err" has meaning to go devs, and the implication of "error" is probably enough for most others, but I could see it coming off as a weird message.
- suggestion: "your docroot %v[docroot] may be incorrect - looking in directory %v[abspath]: %v[err]"

Reply:
On err: Probably if we're going to get a style guide together for error messages we should get one together, and I fully support that.  
I'm always on the debugger's side of this one; in this particular case the user has used a flag that we don't document or suggest. Implementing your suggestion though, which displays the err to end-users anyway :) Since mostly we're the ones that are going to hear about it we need to get this info (especially the err).